modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java

```diff
@@ -1840,24 +1840,18 @@ else if (article != null) {
 
 	@Override
 	public JournalArticle fetchDisplayArticle(long groupId, String articleId) {
-		List<JournalArticle> articles = journalArticlePersistence.findByG_A_ST(
-			groupId, articleId, WorkflowConstants.STATUS_APPROVED);
-
-		if (articles.isEmpty()) {
-			return null;
-		}
+		QueryDefinition<JournalArticle> queryDefinition = new QueryDefinition<>(
+			WorkflowConstants.STATUS_APPROVED, 0, 1, null);
 
 		Date date = new Date();
 
-		for (JournalArticle article : articles) {
-			Date displayDate = article.getDisplayDate();
-			Date expirationDate = article.getExpirationDate();
-
-			if (((displayDate == null) || displayDate.before(date)) &&
-				((expirationDate == null) || expirationDate.after(date))) {
+		List<JournalArticle> articles =
+			journalArticleFinder.filterFindByG_A_LtDD_GtED_ST(
+				groupId, articleId, date, date,
```

Comment:
SF issue

It's weird to pass the same variable twice. Since both are different, we could rename "date" to "now" or use new Date() directly.

Reply:
Renamed to `now`